x/supply/keeper/bank.go

```diff
@@ -0,0 +1,138 @@
+	"github.com/cosmos/cosmos-sdk/x/supply/types"
+)
+
+// SendCoinsFromModuleToAccount trasfers coins from a ModuleAccount to an AccAddress
+func (k Keeper) SendCoinsFromModuleToAccount(ctx sdk.Context, senderModule string,
+	recipientAddr sdk.AccAddress, amt sdk.Coins) sdk.Error {
+
+	senderAddr := k.GetModuleAddress(senderModule)
+	if senderAddr == nil {
+		return sdk.ErrUnknownAddress(fmt.Sprintf("module account %s does not exist", senderModule))
+	}
+
+	return k.bk.SendCoins(ctx, senderAddr, recipientAddr, amt)
+}
+
+// SendCoinsFromModuleToModule trasfers coins from a ModuleAccount to another
+func (k Keeper) SendCoinsFromModuleToModule(ctx sdk.Context, senderModule, recipientModule string, amt sdk.Coins) sdk.Error {
+
+	senderAddr := k.GetModuleAddress(senderModule)
+	if senderAddr == nil {
+		return sdk.ErrUnknownAddress(fmt.Sprintf("module account %s does not exist", senderModule))
+	}
+
+	recipientAddr := k.GetModuleAddress(recipientModule)
+	if recipientAddr == nil {
+		return sdk.ErrUnknownAddress(fmt.Sprintf("module account %s does not exist", recipientModule))
+	}
+
+	return k.bk.SendCoins(ctx, senderAddr, recipientAddr, amt)
+}
+
+// SendCoinsFromAccountToModule trasfers coins from an AccAddress to a ModuleAccount
+func (k Keeper) SendCoinsFromAccountToModule(ctx sdk.Context, senderAddr sdk.AccAddress,
+	recipientModule string, amt sdk.Coins) sdk.Error {
+
+	recipientAddr := k.GetModuleAddress(recipientModule)
+	if recipientAddr == nil {
+		return sdk.ErrUnknownAddress(fmt.Sprintf("module account %s does not exist", recipientModule))
+	}
+
+	return k.bk.SendCoins(ctx, senderAddr, recipientAddr, amt)
+}
+
+// DelegateCoinsFromAccountToModule delegates coins and transfers
+// them from a delegator account to a module account
+func (k Keeper) DelegateCoinsFromAccountToModule(ctx sdk.Context, senderAddr sdk.AccAddress,
+	recipientModule string, amt sdk.Coins) sdk.Error {
+
+	recipientAddr := k.GetModuleAddress(recipientModule)
+	if recipientAddr == nil {
+		return sdk.ErrUnknownAddress(fmt.Sprintf("module account %s does not exist", recipientModule))
+	}
+
+	_, err := k.bk.DelegateCoins(ctx, senderAddr, recipientAddr, amt)
+	return err
+}
+
+// UndelegateCoinsFromModuleToAccount undalegates the unbonding coins and transfers
+// them from a module account to the delegator account
+func (k Keeper) UndelegateCoinsFromModuleToAccount(ctx sdk.Context, senderModule string,
+	recipientAddr sdk.AccAddress, amt sdk.Coins) sdk.Error {
+
+	senderAddr := k.GetModuleAddress(senderModule)
+	if senderAddr == nil {
+		return sdk.ErrUnknownAddress(fmt.Sprintf("module account %s does not exist", senderModule))
+	}
+
+	_, err := k.bk.UndelegateCoins(ctx, senderAddr, recipientAddr, amt)
+	return err
+}
+
+// MintCoins creates new coins from thin air and adds it to the MinterAccount.
+// Panics if the name maps to a HolderAccount
+func (k Keeper) MintCoins(ctx sdk.Context, moduleName string, amt sdk.Coins) sdk.Error {
```

Comment:
Again, non-blocking/for a subsequent PR (I'll create an issue with a brain dump), but having concrete methods such as `MintCoins` and `BurnCoins` is pretty rigid and inflexible for downstream app devs.

tl;dr I foresee the ability to register arbitrary permission**S** with arbitrary function calls for each permission (e.g. burn coins).